pkg/transaction/eventgenerator.go

```diff
@@ -191,6 +200,123 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
 	return events, nil
 }
 
+// getConsumerDetails - get the consumer information to add to transaction event.  If we don't have any
+// 		information we need to get the consumer information, then we just return nil
+func (e *Generator) getConsumerDetails(summaryEvent LogEvent) *ConsumerDetails {
+	cacheManager := agent.GetCacheManager()
+	appName := unknown
+	apiID := ""
+	stage := ""
+
+	if summaryEvent.TransactionSummary.Application != nil {
+		appName = summaryEvent.TransactionSummary.Application.Name
+		e.logger.
+			WithField("appName", appName).
+			Debug("transaction summary application name")
+	}
+
+	// get proxy information
+	if summaryEvent.TransactionSummary.Proxy != nil {
+		apiID = summaryEvent.TransactionSummary.Proxy.ID
+		stage = summaryEvent.TransactionSummary.Proxy.Stage
+		e.logger.
+			WithField("apiID", apiID).
+			WithField("stage", stage).
+			Debug("transaction summary proxy information")
+	} else {
+		return nil
+	}
```

Comment:
Can you switch the parts of this conditional?

if sumEv....Proxy == nil {
return nil
}

Reply:
Will do